android/app/src/main/java/org/openbot/common/ControlsFragment.java

```diff
@@ -211,6 +217,34 @@ private void handlePhoneControllerEvents() {
         );
   }
 
+  class KillSwitch {
```

Comment:
Are you constantly sending some controls to make sure the bot does not get killed by mistake. For example if user drives at a constant speed, there may not be any new event for a while. 

For the communication between robot and bot app, I have implemented a heartbeat (will make a PR soon). Basically every 250ms a heartbeat message will be sent from the phone to the bot. The bot will stop if there is some issue in the communication channel.

Reply:
We send drive commands every 50 ms when the controller is active. The kill switch will trigger if it does not receive drive command in 500 ms.

Reply:
When using sliders to drive, this is usually a non-issue, because the sliders revert back to zero. Using tilt control can leave the bot running after the controller has stopped sending commands. That is why I added this kill switch.

Reply:
Ok sounds good. I have tested everything and it looks good. The one thing is that it is a bit difficult to find the zero point again to stop the car. I was wondering if it makes sense to add a brake button to the screen. Maybe even do throttle control on the screen and only use accelerometer for steering. What are your thoughts on this?

Reply:
As for the kill switch, let's put in into a separate file since it is a class. Or is there a way to do this from the controller? I would prefer not to have this kill switch wrapper.

Reply:
- OK, I will remove the KillSwitch. It is a bit inefficient anyway. My original idea was to have this "Transmission" class wrapper, which can change the driving characteristics, and a killSwitch() being just one method. But I see your point, I will move it to another class and not use it. The only issue is a `runaway bot`. We send a stop command on `disconnect`, but if we change the controller type on the bot app, it will not get a disconnect event. Anyway, I will check this again.

- For the zero point, I think I have a `dead zone` which will send 0,0 if the control is near zero. I will check if this is still there, and maybe widen the range. There is also an option to stop the robot if you place the controller flat on the table at any time, which should work now.

- I will update the PR soon.

Reply:
"I was wondering if it makes sense to add a brake button to the screen. Maybe even do throttle control on the screen and only use accelerometer for steering. What are your thoughts on this?"

Yes, I have seen this in other apps, like "Racing in a Car" app. If you think this is better, we can do this. Full "Zen" mode maybe a bit more difficult to drive.

Reply:
I guess it's hard to know without trying. But I have also seen this in many car racing games. I guess it's easier to control only one axis via rotation.

Reply:
- I added `accelerator` and `break` pedals to the tilt screen. These are implemented in ```Accelerator.kt``` and ```Break.kt``` files. The icons are very basic, but if somebody can draw a better pedals, please go ahead and replace them.
- In both these file, there is a class called `DrivingCharacteristics`, which defined various parameters controlling responsiveness of the pedals. One can experiment with these to archive optimum driving experience.